pkg/matrix-diff/matrix-diff.go

```diff
@@ -0,0 +1,75 @@
+package matrixdiff
+
+import (
+	"fmt"
+
+	"github.com/openshift-kni/commatrix/pkg/types"
+)
+
+// Describes the status of an endpointslice.
+type epsStatus int
+
+// Used = the port is open on the node.
+// NotUsed = the port is not open on the node.
+// Missing = no endpointslice defined for an open port.
+const (
+	used epsStatus = iota
+	notUsed
+	missing
+)
+
+type MatrixDiff struct {
+	matrix        []types.ComDetails
+	cdToEPSStatus map[string]epsStatus
+}
+
+// Generates the diff between mat1 to mat2.
+func Generate(mat1 *types.ComMatrix, mat2 *types.ComMatrix) (MatrixDiff, error) {
```

Comment:
its not better to have  func (m *ComMatrix) Generate(other *ComMatrix) (MatrixDiff, error){}?
like from one matrix we are generating diff from the other on that format its more clear

Reply:
To do that you have to keep this function in the types files (where ComMatrix is defined) which is too random and general (and should be gone soon) so I prefer creating a new package for diff